vespajlib/src/main/java/com/yahoo/lang/CachedSupplier.java

```diff
@@ -0,0 +1,52 @@
+// Copyright Verizon Media. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+package com.yahoo.lang;
+
+import java.time.Clock;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.function.Supplier;
+
+/**
+ * Supplier that caches the value for a given duration with ability to invalidate at demand.
+ * Is thread safe.
+ *
+ * @author freva
+ */
+public class CachedSupplier<T> implements Supplier<T> {
```

Comment:
Or `MemoizedSupplier`?
I also considered having 2 factory methods similar to Guavas `Suppliers`: `memoize()` and `memoizeWithExpiration()`, but I'm not sure if we ever need a simple memoized supplier that needs refreshing....

Reply:
The main thing is ability to invalidate instead of having to create new Supplier, another bonus is that `CachedSupplier` is a `java.util.function.Supplier`, while Guava returns `com.google.common.base.Supplier`.